language/expressions.md

```diff
@@ -720,9 +726,26 @@ result. Chained assignments are permitted.
 
 Examples:
 
+~~~puppet
+
     $a = 10
     $x = $y = 0
 
+    # from array
+    [$a, $b] = [1, 2]  # assigns 1 to $a, and 2 to $b
+
+    # from hash
+    [$a, $b] = { a => 10, b => 20, c => 30 }  # assigns 10 to $a, and 20 to $b
+
+    # from class
+    class mymodule::someclass::example($x = 100) {
+      $a = 10
+    }
+    include example
+    [$a, $x] = Class['mymodule::someclass::example']  # assigns 10 to $a, and 100 to $x
```

Comment:
You can get parameters from a define with `X[meh][x]`  but only if it has been evaluated. We could add support for multi-assign for that, but people will probably be bit by evaluation order all the time trying to use that. Also, since the parameters to the defines must come from somewhere - and that is usually from classes; seems better to get them from there (i.e. from the input, rather from the "resulting resource").